circleci/images/stylechecker/Dockerfile

```diff
@@ -1,22 +1,41 @@
-FROM alpine:3.9
+FROM alpine:latest
 ARG TOOLS_VERSION
 
-RUN apk add --no-cache --virtual installdeps curl make sed && \
-    apk add --no-cache \
-        ca-certificates \
-        bash \
-        git \
-        grep \
-        gzip \
-        openssh \
-        perl \
-        tar \
-        uncrustify && \
-    curl -sfLO "https://github.com/citusdata/tools/archive/v${TOOLS_VERSION}.tar.gz" && \
-    tar xzf "v${TOOLS_VERSION}.tar.gz" && \
-    cd "tools-${TOOLS_VERSION}" && \
-    sed -i -e's/align_number_left/align_number_right/g' uncrustify/citus-style.cfg && \
-    make uncrustify/.install && \
-    cd .. && \
-    rm -rf "citus-${VERSION}" "v${TOOLS_VERSION}.tar.gz" && \
-    apk del installdeps
+COPY ./files/etc/requirements.txt /tmp/etc/
+RUN apk add --no-cache --virtual installdeps \
+      curl \
+      make \
+      cmake \
+      gcc \
+      g++ \
+      musl-dev \
+      python3-dev \
+  && apk add --no-cache \
+      ca-certificates \
+      cmake \
+      bash \
+      git \
+      grep \
+      gzip \
+      musl-dev \
+      openssh \
+      perl \
+      python3 \
+      py3-pip \
+      tar \
```

Comment:
I really like the effort to delete all install dependencies and leave only the packages that will be used in the package. However, I do not really get how and why some of the packages that are not listed under the virtual `installdeps` are used in the actual image. For example can we move the following to `installdeps` virtual package:
- `cmake` appears to be used only when building uncrustify
- `gzip` is used when extracting uncrustify tarball
- `musl-dev` is a standard C library. I do not really know why we need it at all
- `openssh` does not look like used at all as we use https connections for github and I do not see any other place that can actually use ssh.
- `py3-pip` is used only here to install packages. Is it not safe to delete it after we are done with installing python packages? Deleting this package may also delete the python packages, so I am not sure if it is safe to move it to virtual package list at all.
- `tar` is used when extracting uncrustify tarball.

Reply:
Moved all of these packages to installdeps. Two callouts:
1. I added py3-packaging to regular installed packages. Since py3-pip pulled that in and it's needed to run our linters due to pyproject.toml.
2. musl-dev is needed during build because some python packages need it to compile some C stuff.

Reply:
Okay added back openssh. It's necessary for circleci to check out the github repo.